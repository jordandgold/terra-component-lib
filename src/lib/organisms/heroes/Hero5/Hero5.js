import React from "react";
import "./Hero5.scss";
import PropTypes from "prop-types";
import CTALinkSection from "../../../CTALinkSection/CTALinkSection";

const Hero5 = props => {
  const { title, text, ctas, image, subtitle } = props;
  return (
    <section className="ter-hero-five">
      <div className="ter-hero-five__image-container">
        <img
          className="ter-hero-five__image"
          src={image.url}
          alt={image.altText}
        />
      </div>
      <div className="ter-hero-five__content-container">
        <h2 className="ter-hero-five__title">{title}</h2>
        {subtitle && <h3 className="ter-hero-five__subtitle">{subtitle}</h3>}
        {text && <p className="ter-hero-five__text">{text}</p>}
        <CTALinkSection ctas={ctas} />
      </div>
    </section>
  );
};

export default Hero5;

Hero5.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  text: PropTypes.string,
  ctas: PropTypes.shape({
    ctaOne: PropTypes.shape({
      className: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired,
      link: PropTypes.string.isRequired
    }),
    ctaTwo: PropTypes.shape({
      className: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired,
      link: PropTypes.string.isRequired
    }),
    subCTA: PropTypes.shape({
      text: PropTypes.string.isRequired,
      link: PropTypes.string.isRequired
    })
  })
};
