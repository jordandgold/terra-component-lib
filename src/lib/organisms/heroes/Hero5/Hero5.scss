@import "../../../global-styles/terra.scss";

.ter-hero-five {
  @media (min-width: $breakpoint-sm) {
    @include grid-container();
    align-items: center;
  }

  &__image-container {
    @media (min-width: $breakpoint-sm) {
      @include grid-column(8);
      padding: spacing("super") spacing("md") spacing("super") spacing("md");
    }

    @media (min-width: $breakpoint-md) {
      @include grid-column(7);
    }
  }

  &__image {
    width: 100%;
  }

  &__content-container {
    margin: spacing("md");

    @media (min-width: $breakpoint-sm) {
      @include grid-column(4);
      margin: 0;
      padding-right: spacing("lg");
    }

    @media (min-width: $breakpoint-md) {
      @include grid-column(5);
    }
  }

  &__title {
    font-size: font-size("lg");

    @media (min-width: $breakpoint-md) {
      font-size: font-size("xxl");
    }
  }

  &__subtitle {
    font-weight: font-weight("light");
    color: color("gray-6");
    margin-top: spacing("md");

    @media (min-width: $breakpoint-md) {
      font-size: font-size("lg");
    }
  }

  &__text {
    margin-top: spacing("md");

    @media (min-width: $breakpoint-md) {
      font-size: font-size("md");
    }
  }

  .ter-cta-section {
    @media (min-width: $breakpoint-xs) {
      width: 175px;
    }
  }

  .ter-button {
    text-align: center;
    display: block;
  }

  .ter-button--secondary--1 {
    margin-top: spacing("xs");
  }

  .ter-cta-section__link {
    margin-top: spacing("mini");
    display: block;
    text-align: center;
  }
}
