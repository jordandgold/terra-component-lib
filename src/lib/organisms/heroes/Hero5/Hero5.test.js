import React from "react";
import { shallow } from "enzyme";
import Hero5 from "./Hero5";

describe("Hero5", () => {
  let wrapper;
  const mockCTALinks = {
    ctaOne: {
      className: "ter-button--primary--1",
      text: "Button",
      link: "#"
    },
    ctaTwo: {
      className: "ter-button--secondary--1",
      text: "Button",
      link: "#"
    },
    subCTA: {
      text: "Button",
      link: "#"
    }
  };

  beforeEach(() => {
    wrapper = shallow(
      <Hero5
        title="Design is design."
        text="Esse aliquip ad in et ut ipsum paEu elit consectetur aliquip excepteur fugiat ut qui dolor pariatur consectetur.riatur elit quis."
        ctas={mockCTALinks}
        image={{ url: "test", altText: "test" }}
      />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should match the snapshot with a subtitle", () => {
    wrapper = shallow(
      <Hero5
        title="Design is design."
        subtitle="I am a subtitle"
        text="Esse aliquip ad in et ut ipsum paEu elit consectetur aliquip excepteur fugiat ut qui dolor pariatur consectetur.riatur elit quis."
        ctas={mockCTALinks}
        image={{ url: "test", altText: "test" }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
