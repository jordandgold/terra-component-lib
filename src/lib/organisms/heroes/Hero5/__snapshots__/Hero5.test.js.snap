// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero5 should match the snapshot 1`] = `
<section
  className="ter-hero-five"
>
  <div
    className="ter-hero-five__image-container"
  >
    <img
      alt="test"
      className="ter-hero-five__image"
      src="test"
    />
  </div>
  <div
    className="ter-hero-five__content-container"
  >
    <h2
      className="ter-hero-five__title"
    >
      Design is design.
    </h2>
    <p
      className="ter-hero-five__text"
    >
      Esse aliquip ad in et ut ipsum paEu elit consectetur aliquip excepteur fugiat ut qui dolor pariatur consectetur.riatur elit quis.
    </p>
    <CTALinkSection
      ctas={
        Object {
          "ctaOne": Object {
            "className": "ter-button--primary--1",
            "link": "#",
            "text": "Button",
          },
          "ctaTwo": Object {
            "className": "ter-button--secondary--1",
            "link": "#",
            "text": "Button",
          },
          "subCTA": Object {
            "link": "#",
            "text": "Button",
          },
        }
      }
    />
  </div>
</section>
`;

exports[`Hero5 should match the snapshot with a subtitle 1`] = `
<section
  className="ter-hero-five"
>
  <div
    className="ter-hero-five__image-container"
  >
    <img
      alt="test"
      className="ter-hero-five__image"
      src="test"
    />
  </div>
  <div
    className="ter-hero-five__content-container"
  >
    <h2
      className="ter-hero-five__title"
    >
      Design is design.
    </h2>
    <h3
      className="ter-hero-five__subtitle"
    >
      I am a subtitle
    </h3>
    <p
      className="ter-hero-five__text"
    >
      Esse aliquip ad in et ut ipsum paEu elit consectetur aliquip excepteur fugiat ut qui dolor pariatur consectetur.riatur elit quis.
    </p>
    <CTALinkSection
      ctas={
        Object {
          "ctaOne": Object {
            "className": "ter-button--primary--1",
            "link": "#",
            "text": "Button",
          },
          "ctaTwo": Object {
            "className": "ter-button--secondary--1",
            "link": "#",
            "text": "Button",
          },
          "subCTA": Object {
            "link": "#",
            "text": "Button",
          },
        }
      }
    />
  </div>
</section>
`;
